### PR TITLE
BUG make sure split keys are handled properly in recipe parser

### LINKED ISSUE #1499: The `skip` line in `meta.yaml` is messed up every time when the bot sends a PR
See https://github.com/conda-forge/cupy-feedstock/pull/160#discussion_r788975169. It happened several times. The bot keeps changing this line
```
skip: true  # [(not win64 and not linux64 and (ppc64le and cuda_compiler_version != "10.2") and not (aarch64 and arm_variant_type == "sbsa")) or cuda_compiler_version in (undefined, "None")]
```
to this nonsense
```
  ? skip__###conda-selector###__(not win64 and not linux64 and (ppc64le and cuda_compiler_version != "10.2") and not (aarch64 and arm_variant_type == "sbsa")) or cuda_compiler_version in (undefined, "None")
  : true
```
causing linter and builder failures.

### PR DESCRIPTION
**Pull request by beckermr**
This PR fixes the issue @leofang noticed in #1499. 

The approach I have taken is to fix only this edge case when

1. A selector was munged
2. The key-value pair is split across two lines
3. the yaml parser used the `?` format for keys. See Wikipedia and yeah this is weird!

Hopefully, the code is specific enough to not cause other issues later.

Closes #1499

### CODE REVIEW

**Comment by beckermr** [comment]
Test failures are unrelated and due to flakey stuff. Going to merge.

**Comment by jakirkham** [comment]
Thanks Matt! 😀

File: conda_forge_tick/recipe_parser/_parser.py [hunk 1]
```diff
@@ -176,6 +178,33 @@ def _unmunge_line(line: str) -> str:
         return line
 
 
+def _unumge_split_key_value_pairs_with_selectors(lines):
```
Comment (isuruf) [inline conda_forge_tick/recipe_parser/_parser.py:181]: unumge -> unmunge?

Comment (beckermr) [reply conda_forge_tick/recipe_parser/_parser.py:181]: Yeah woops! Will fix and thank you!

**Comment by leofang** [comment]
@beckermr I confirm it's fixed! https://github.com/conda-forge/cupy-feedstock/pull/165

**Comment by beckermr** [comment]
Awesome!

**Pull request merged**